secboot/secboot_hooks.go

```diff
@@ -160,14 +168,26 @@ func ResealKeysWithFDESetupHook(keyFiles []string, primaryKeyFile string, models
 		}
 
 		writer := sb.NewFileKeyDataWriter(keyFile)
-		if err := keyData.WriteAtomic(writer); err != nil {
+		if err := loadedKey.KeyData.WriteAtomic(writer); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
 
+func unlockDiskWithHookV1Key(mapperName, sourceDevice string, sealed []byte) error {
```

Comment:
tests don't reach here

Reply:
The tests for it were botched. Also I have noticed that TestUnlockVolumeUsingSealedKeyIfEncryptedFdeRevealKeyV1AndV1GeneratedSealedKeyFile and TestUnlockVolumeUsingSealedKeyIfEncryptedFdeRevealKeyV1 were almost the same, just some checks were different. So I also kept only one with all the checks.